com.unity.multiplayer.mlapi/Runtime/Messaging/RpcQueue/RpcQueueProcessor.cs

```diff
@@ -100,67 +109,92 @@ public void ProcessSendQueue()
         }
 
         /// <summary>
-        ///  QueueInternalMLAPICommand
-        ///  Added this as an example of how to add internal messages to the outbound send queue
+        ///  Used to queue lower layer MLAPI commands
+        ///  Currently only used for spawning and destroying objects
         /// </summary>
         /// <param name="queueItem">message queue item to add<</param>
-        public void QueueInternalMLAPICommand(RpcFrameQueueItem queueItem)
+        internal void QueueInternalMLAPICommand(RpcFrameQueueItem queueItem)
         {
             m_InternalMLAPISendQueue.Add(queueItem);
         }
 
         /// <summary>
         /// Generic Sending Method for Internal Messages
-        /// TODO: Will need to open this up for discussion, but we will want to determine if this is how we want internal MLAPI command
-        /// messages to be sent.  We might want specific commands to occur during specific network update regions (see NetworkUpdate
+        /// If a client is not finished loading the current scene, then the messages will be kept until
+        /// the clients scene loading status is marked as completed.
         /// </summary>
-        public void InternalMessagesSendAndFlush()
+        internal void InternalMessagesSendAndFlush()
         {
-            foreach (RpcFrameQueueItem queueItem in m_InternalMLAPISendQueue)
+            if(NetworkManager.Singleton && NetworkManager.Singleton.IsListening)
```

Comment:
again, similar to above, I'm not sure if this check is really needed. maybe `IsListening` but not sure if `Singleton` check is required here. also, why we're checking if we're `IsListening` here? shouldn't the call-site of `InternalMessagesSendAndFlush` method do that check first before calling this method in the first place? I have questions... :P